// @wordpress/block-editor/build-module/components/inserter/reusable-block-rename-hint.js
import { createElement } from "@wordpress/element";

/**
 * WordPress dependencies
 */
import { Button } from '@wordpress/components';
import { useDispatch, useSelect } from '@wordpress/data';
import { focus } from '@wordpress/dom';
import { useRef } from '@wordpress/element';
import { __ } from '@wordpress/i18n';
import { close } from '@wordpress/icons';
import { store as preferencesStore } from '@wordpress/preferences';
const PREFERENCE_NAME = 'isResuableBlocksrRenameHintVisible';
export default function ReusableBlocksRenameHint() {
  const isReusableBlocksRenameHint = useSelect(select => {
    var _select$get;

    return (_select$get = select(preferencesStore).get('core', PREFERENCE_NAME)) !== null && _select$get !== void 0 ? _select$get : true;
  }, []);
  const ref = useRef();
  const {
    set: setPreference
  } = useDispatch(preferencesStore);

  if (!isReusableBlocksRenameHint) {
    return null;
  }

  return createElement("div", {
    ref: ref,
    className: "reusable-blocks-menu-items__rename-hint"
  }, createElement("div", {
    className: "reusable-blocks-menu-items__rename-hint-content"
  }, __('Reusable blocks are now called patterns. A synced pattern will behave in exactly the same way as a reusable block.')), createElement(Button, {
    className: "reusable-blocks-menu-items__rename-hint-dismiss",
    icon: close,
    iconSize: "16",
    label: __('Dismiss hint'),
    onClick: () => {
      // Retain focus when dismissing the element.
      const previousElement = focus.tabbable.findPrevious(ref.current);
      previousElement?.focus();
      setPreference('core', PREFERENCE_NAME, false);
    },
    showTooltip: false
  }));
}
