// @wordpress/block-editor/build/components/spacing-sizes-control/input-controls/single.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = SingleInputControl;

var _element = require("@wordpress/element");

var _spacingInputControl = _interopRequireDefault(require("./spacing-input-control"));

var _utils = require("../utils");

/**
 * Internal dependencies
 */
function SingleInputControl({
  minimumCustomValue,
  onChange,
  onMouseOut,
  onMouseOver,
  showSideInLabel,
  side,
  spacingSizes,
  type,
  values
}) {
  const createHandleOnChange = currentSide => next => {
    const nextValues = { ...values
    };
    nextValues[currentSide] = next;
    onChange(nextValues);
  };

  return (0, _element.createElement)(_spacingInputControl.default, {
    label: _utils.LABELS[side],
    minimumCustomValue: minimumCustomValue,
    onChange: createHandleOnChange(side),
    onMouseOut: onMouseOut,
    onMouseOver: onMouseOver,
    showSideInLabel: showSideInLabel,
    side: side,
    spacingSizes: spacingSizes,
    type: type,
    value: values[side],
    withInputField: false
  });
}
